sentry-android-core/src/main/java/io/sentry/android/core/AndroidContinuousProfiler.java

```diff
@@ -27,7 +31,8 @@
 import org.jetbrains.annotations.VisibleForTesting;
 
 @ApiStatus.Internal
-public class AndroidContinuousProfiler implements IContinuousProfiler {
+public class AndroidContinuousProfiler
+    implements IContinuousProfiler, RateLimiter.IRateLimitObserver {
```

Comment:
do you also want to observe network connectivity actually? In case the device gets offline in the middle of a profile running, we'd still overflow the cache I guess?

Reply:
I do the check when each profile chunk starts. 
I just want to avoid the case where the device goes offline for 1 second and then reconnects. In this case we would drop the whole profile
And the overhead is low enough not to "care" about stopping early when device is offline

Reply:
ah yeah, I forgot that we call start() continuously, that makes sense then!